File: yuiextz/src/org/zkoss/yuiext/layout/South.java
/* South.java

 {{IS_NOTE
 Purpose:
 
 Description:
 
 History:
 Aug 7, 2007 6:28:04 PM , Created by jumperchen
 }}IS_NOTE

 Copyright (C) 2007 Potix Corporation. All Rights Reserved.

 {{IS_RIGHT
 This program is distributed under GPL Version 2.0 in the hope that
 it will be useful, but WITHOUT ANY WARRANTY.
 }}IS_RIGHT
 */
package org.zkoss.yuiext.layout;

/**
 * This component is a south region.
 * 
 * @author jumperchen
 * 
 */
public class South extends SplitLayoutRegion {

	private int[] _cmargins = new int[] { 2, 0, 0, 2 };

	/**
	 * defaults to {top: 2, left: 0, right: 0, bottom: 2}
	 */
	public int[] getCmargins() {
		return _cmargins;
	}

	public void setCmargins(int top, int left, int right, int bottom) {
		boolean f = false;
		if (_cmargins[0] != top) {
			f = true;
			_cmargins[0] = top;
		}
		if (_cmargins[1] != left) {
			f = true;
			_cmargins[1] = left;
		}
		if (_cmargins[2] != right) {
			f = true;
			_cmargins[2] = right;
		}
		if (_cmargins[3] != bottom) {
			f = true;
			_cmargins[3] = bottom;
		}
		if (f)
			smartUpdate("z.cmargins", _cmargins[0] + "," + _cmargins[1] + ","
					+ _cmargins[2] + "," + _cmargins[3]);
	}

	/**
	 * Returns {@link BasicLayoutRegion#POSITION_SOUTH}
	 */
	public String getPosition() {
		return POSITION_SOUTH;
	}

}

File: zul/src/main/resources/web/js/zul/layout/South.ts
/* South.ts

	Purpose:

	Description:

	History:
		Wed Jan  7 12:14:58     2009, Created by jumperchen

Copyright (C) 2008 Potix Corporation. All Rights Reserved.

This program is distributed under LGPL Version 2.1 in the hope that
it will be useful, but WITHOUT ANY WARRANTY.
*/
/**
 * A south region of a border layout.
 * <p>Default {@link #getZclass}: z-south.
 */
@zk.WrapClass('zul.layout.South')
export class South extends zul.layout.LayoutRegion {
	_sumFlexHeight = true; //indicate shall add this flex height for borderlayout. @See _fixMinFlex in widget.js
	override _cmargins = [0, 0, 0, 0];
	override sanchor = 'b';

	/**
	 * The width can't be specified in this component because its width is
	 * determined by other region components ({@link West} or {@link East}).
	 * @param String width
	 */
	override setWidth(width: string): this { // readonly
		return this;
	}

	/**
	 * Returns {@link Borderlayout#SOUTH}.
	 * @return String
	 */
	override getPosition(): string {
		return zul.layout.Borderlayout.SOUTH;
	}

	/**
	 * Returns the size of this region. This method is shortcut for
	 * {@link #getHeight()}.
	 * @return String
	 */
	getSize(): string | null | undefined {
		return this.getHeight();
	}

	/**
	 * Sets the size of this region. This method is shortcut for
	 * {@link #setHeight(String)}.
	 * @param String size
	 */
	setSize(size: string): this {
		return this.setHeight(size);
	}

	override _ambit2(ambit: zul.layout.LayoutRegionAmbit, mars: zk.Dimension, split: { offsetWidth; offsetHeight }): void {
		ambit.w = mars.width;
		ambit.h += split.offsetHeight;
		ambit.ts = ambit.y + ambit.h + (mars.height - mars.top); // total size;
		ambit.y = ambit.h + (mars.height - mars.top);
	}

	override _reszSp2(ambit: zul.layout.LayoutRegionAmbit, split: { w; h }): Partial<{ left; top; width; height }> {
		ambit.h -= split.h;
		ambit.y += split.h;
		return {
			left: jq.px0(ambit.x),
			top: jq.px0(ambit.y - split.h),
			width: jq.px0(ambit.w)
		};
	}
}